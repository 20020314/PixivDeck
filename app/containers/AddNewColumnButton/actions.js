// @flow
import type { User } from '../../types/user'
import type { Action } from './actionTypes.js'
import { ADD_USER_ILLUST } from './constants'

export function addUserIllust(user: User): Action {
  return {
    type: ADD_USER_ILLUST,
    user,
  }
}
