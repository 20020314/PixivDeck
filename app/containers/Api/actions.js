// @flow
import type { Response } from 'services/api'

import { API_REQUEST_SUCCESS } from './actionTypes'
import type { ApiRequestSuccess } from './actionTypes'

export function apiRequestSuccess(response: Response): ApiRequestSuccess {
  return {
    type: API_REQUEST_SUCCESS,
    response,
  }
}
