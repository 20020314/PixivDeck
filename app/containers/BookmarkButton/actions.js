// @flow
import type { Restrict } from './types'

import {
  ADD_BOOKMARK_REQUEST,
  ADD_BOOKMARK_SUCCESS,
  ADD_BOOKMARK_FAILURE,
  DELETE_BOOKMARK_REQUEST,
  DELETE_BOOKMARK_SUCCESS,
  DELETE_BOOKMARK_FAILURE,
} from './actionTypes'
import type {
  AddBookmarkRequest,
  AddBookmarkSuccess,
  AddBookmarkFailure,
  DeleteBookmarkRequest,
  DeleteBookmarkSuccess,
  DeleteBookmarkFailure,
} from './actionTypes'

export function addBookmarkRequest(
  id: number,
  restrict: Restrict
): AddBookmarkRequest {
  return {
    type: ADD_BOOKMARK_REQUEST,
    id,
    restrict,
  }
}
export function addBookmarkSuccess(
  id: number,
  restrict: Restrict
): AddBookmarkSuccess {
  return {
    type: ADD_BOOKMARK_SUCCESS,
    id,
    restrict,
  }
}
export function addBookmarkFailure(
  id: number,
  error: string
): AddBookmarkFailure {
  return {
    type: ADD_BOOKMARK_FAILURE,
    id,
    error,
  }
}
export function deleteBookmarkRequest(id: number): DeleteBookmarkRequest {
  return {
    type: DELETE_BOOKMARK_REQUEST,
    id,
  }
}
export function deleteBookmarkSuccess(id: number): DeleteBookmarkSuccess {
  return {
    type: DELETE_BOOKMARK_SUCCESS,
    id,
  }
}
export function deleteBookmarkFailure(
  id: number,
  error: string
): DeleteBookmarkFailure {
  return {
    type: DELETE_BOOKMARK_FAILURE,
    id,
    error,
  }
}
