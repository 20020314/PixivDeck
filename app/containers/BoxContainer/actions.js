// @flow
import type { Action } from './actionTypes.js'
import { OPEN_PIXIV } from './constants'

export function openPixiv(id: number): Action {
  return {
    type: OPEN_PIXIV,
    id,
  }
}
