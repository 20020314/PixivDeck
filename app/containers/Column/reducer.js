// @flow
import update from 'utils/update'

export type BaseColumn = {
  illustIds: Array<number>,
  nextUrl: ?string,
}

type ActionTypes = {
  ADD_COLUMN_SUCCESS: $Subtype<string>,
  SET_NEXT_URL: $Subtype<string>,
}

export function baseReducer<
  T: BaseColumn,
  S: { [*]: T },
  AS: ActionTypes,
  A: { +type: $Subtype<string> }
>(types: AS, state: S, action: A): S {
  switch (action.type) {
    case types.SET_NEXT_URL:
      return update(state, action, { nextUrl: action.nextUrl })
    case types.ADD_COLUMN_SUCCESS:
      return update(state, action, { illustIds: [], nextUrl: null })
    default:
      return state
  }
}
