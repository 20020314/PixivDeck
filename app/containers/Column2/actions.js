// @flow
import type { Action } from './actionTypes.js'
import { ADD_COLUMN } from './constants'
import type { Body } from './reducer'

export function addColumn(id: string, body: Body): Action {
  return {
    type: ADD_COLUMN,
    id,
    body,
  }
}
