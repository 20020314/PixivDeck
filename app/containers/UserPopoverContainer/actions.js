// @flow
import type { Action } from './actionTypes.js'
import { OPEN_USER_POPOVER } from './constants'

export function openUserPopover(id: number): Action {
  return {
    type: OPEN_USER_POPOVER,
    id,
  }
}
