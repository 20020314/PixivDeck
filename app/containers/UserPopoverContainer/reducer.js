// @flow
import type { Illust } from 'types/illust'
import type { Action } from './actionTypes'
import * as Actions from './constants'

export type State = {
  illusts: Array<Illust>,
}

const initialState: State = {
  illusts: [],
}

export default function(state: State = initialState, action: Action): State {
  switch (action.type) {
    case Actions.ADD_POPOVER:
      return { illusts: action.illusts }
    case Actions.CLEAR_POPOVER:
      return { illusts: [] }
    default:
      return state
  }
}
