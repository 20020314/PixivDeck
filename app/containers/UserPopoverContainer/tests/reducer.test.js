// @flow
import reducer from '../reducer'
import * as actions from '../actions'

const df = {
  illusts: [],
}

test('default action', () => {
  // $FlowFixMe
  expect(reducer(df, { type: 'default action' })).toMatchSnapshot()
})

test('POPOVER_SUCCESS', () => {
  expect(reducer(df, actions.popoverSuccess([1, 2, 3]))).toMatchSnapshot()
})

test('CLEAR', () => {
  const s = { illusts: [1, 2, 3] }
  expect(reducer(s, actions.clear())).toMatchSnapshot()
})
