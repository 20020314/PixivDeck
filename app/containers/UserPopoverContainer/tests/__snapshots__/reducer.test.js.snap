// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`CLEAR 1`] = `
Object {
  "illusts": Array [],
}
`;

exports[`POPOVER_SUCCESS 1`] = `
Object {
  "illusts": Array [
    1,
    2,
    3,
  ],
}
`;

exports[`default action 1`] = `
Object {
  "illusts": Array [],
}
`;
